calculator.rb

```diff
@@ -0,0 +1,98 @@
+#Setting Up Operators
+operators = ["1", "add", "+", "2", "subtract", "-", "3", "multiply", "*", "4", "divide", "/", "5", "exponent", "^", "6", "modulo", "remainder", "%"]
+numbers = []
+
+#Validation Methods
+def data_type (num_one)
+    if num_one.include?(".") == true
+        num_one = num_one.to_f
+    else
+        num_one = num_one.to_i
+    end
+    return num_one
+end
+
+def number?(obj)
+    obj = obj.to_s unless obj.is_a? String
+    /\A[+-]?\d+(\.[\d]+)?\z/.match obj
+    #Lines 15-17 Source: https://medium.com/launch-school/number-validation-with-regex-ruby-393954e46797
```

Comment:
Thanks for referencing code that you found! We will learn more about regex expressions soon.